events/api.py

```diff
@@ -3025,15 +3025,21 @@ def _filter_event_queryset(queryset, params, srs=None):  # noqa: C901
         queryset = queryset.filter(publication_status=PublicationStatus.PUBLIC)
 
     # Filter by event status
-    val = params.get("event_status", None)
-    if val and val.lower() == "eventscheduled":
-        queryset = queryset.filter(event_status=Event.Status.SCHEDULED)
-    elif val and val.lower() == "eventrescheduled":
-        queryset = queryset.filter(event_status=Event.Status.RESCHEDULED)
-    elif val and val.lower() == "eventcancelled":
-        queryset = queryset.filter(event_status=Event.Status.CANCELLED)
-    elif val and val.lower() == "eventpostponed":
-        queryset = queryset.filter(event_status=Event.Status.POSTPONED)
+    statuses = params.get("event_status", None)
+
+    if statuses:
+        searched_statuses = []
+        for val in statuses.split(","):
+            if val and val.lower() == "eventscheduled":
+                searched_statuses.append(Event.Status.SCHEDULED)
+            elif val and val.lower() == "eventrescheduled":
+                searched_statuses.append(Event.Status.RESCHEDULED)
+            elif val and val.lower() == "eventcancelled":
+                searched_statuses.append(Event.Status.CANCELLED)
+            elif val and val.lower() == "eventpostponed":
+                searched_statuses.append(Event.Status.POSTPONED)
+
+        queryset = queryset.filter(event_status__in=searched_statuses)
```

Comment:
Something like
```
if status_param := params.get("event_status"):
    statuses = []
    for value in (s.lower() for s in status_param.split(",")):
        if value == "eventscheduled":
            statuses.append(Event.Status.SCHEDULED)
        elif value == "eventrescheduled":
            statuses.append(Event.Status.RESCHEDULED)
        elif value == "eventcancelled":
            statuses.append(Event.Status.CANCELLED)
        elif value == "eventpostponed":
            statuses.append(Event.Status.POSTPONED)

    queryset = queryset.filter(event_status__in=statuses)
```
could be a bit more pythonic and DRY.

Reply:
Added note to the documentation and changed code as suggested.